hudi-common/src/main/java/org/apache/hudi/common/util/collection/FilterIterator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util.collection;
+
+import org.apache.hudi.common.util.ValidationUtils;
+
+import java.util.Iterator;
+import java.util.function.Predicate;
+
+public class FilterIterator<R> implements Iterator<R> {
+
+  private final Iterator<R> source;
+
+  private final Predicate<R> filter;
+
+  private R current;
+
+  private boolean init;
+
+  public FilterIterator(Iterator<R> source, Predicate<R> filter) {
+    this.source = source;
+    this.filter = filter;
+  }
+
+  private void init() {
+    if (!init) {
+      while (source.hasNext()) {
+        R next = source.next();
+        if (filter.test(next)) {
+          current = next;
+          break;
+        }
+      }
+      init = true;
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    init();
```

Comment:
Looks like the `init()` can be elimimated.

Reply:
> Looks like the `init()` can be elimimated.

Nice catch~ Changed.